Translations

PogSmok opened the pull request:
Wyczekiwane tłumaczenia z JSON-ów.

Należy jeszcze sprawdzić komendy inhousowe.

PogSmok committed:
Create EN.json

PogSmok committed:
Create PL.json

PogSmok committed:
Change hardcoded answers for JSON imports

PL.json - polish translations
EN.json - english defaults
Both files must be in the same folder as discord_bot.py, otherwise their paths must be changed (line 35 and 39)

PogSmok committed:
Format fix

sellmark commented on app/balancer/management/commands/discord_bot.py:
```diff
@@ -27,6 +27,18 @@
+    english = json.load(english_file)
+    TRANSLATIONS["EN"] = english
+
+with open(os.path.join(os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__))), "PL.json"), encoding='utf-8') as polish_file:
```
Hej, tu nie przesadziłeś z odgadywaniem tej ścieżki? :D

sellmark replied:
W sensie - nie idzie krócej tego zrobić?

PogSmok replied:
Nie, używanie po prostu "EN.json" lub "PL.json" jako ścieżki nie działa zawsze i jest niezgodne z dobrymi praktykami.  W zależności od systemu operacyjnego oraz sposobu uruchamiania skryptu, python może inaczej zinterpretować ścieżkę i zwracać błędy.

sellmark replied:
Oczywiście, nie każę Ci brnąć w taką skrajność, ale zdaje się w projekcie mamy już złapany root dir projektu pod zmienną BASE_DIR

Mowię o pliku settings.py, gdzie jest BASE_DIR zdefiniowany i tam w paru miejscach jest os.path.join na tym:

na przykład: 

` 'LOCATION': os.path.join(BASE_DIR, 'cache'),`

sellmark commented on app/balancer/management/commands/discord_bot.py:
```diff
@@ -27,6 +27,18 @@
 
+import json
+#importing translations
+#ideally this should be in a separate file/folder for translations, but since file structure in this project doesn't exist... :)
```
Mamy możliwość stworzyć folder, np w `app/balancer/translations` i nawet wrzucić ten path gdzieś podczas INIT-u bota, aby już znał tą ścieżkę i nie trzeba było jej tak odgadywać jak niżej.

PogSmok replied:
Dodałem w app/balancer/management/commands/command_translation

sellmark replied:
możemy równie dobrze dać do root projektu, nic nam nie szkodzi i potem wykorzystać powyżej poruszony BASE_DIR

` 'LOCATION': os.path.join(BASE_DIR, 'translations'),`

po co później szukać tak głęboko tego jsona, zgodzisz się?

sellmark commented on app/balancer/management/commands/discord_bot.py:
```diff
@@ -27,6 +27,18 @@
+#ideally this should be in a separate file/folder for translations, but since file structure in this project doesn't exist... :)
+TRANSLATIONS = {}
+LANG = "PL"
+with open(os.path.join(os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__))), "EN.json"), encoding='utf-8') as english_file:
```
moglibyśmy tutaj zrobić funkcję "getTranslation("PL")" która zwróciłaby nam to z jsona w arrayce, czyli byłoby tak:

TRANSLATIONS["EN"] = self.getTranslations("EN");

Poza tym, mój zamysł był taki, aby aplikacja obsługiwałą tylko jeden plik json na raz. I to jest odpowiedzialnością programisty potem jak coś robi, żeby zmienić to np. gdzieś w konfigu albo w .env jaki jest domyślny język i na nim leci już cała apka dalej. 

Mogę nie mieć racji, więc jakie jest Twoje zdanie?

PogSmok replied:
Robienie funkcji to jest trochę lanie wody, już i tak zczytujemy ze słownika po zmiennej. Jeżeli o konfigurację chodzi, to zmienna LANG istnieje. Uważam, że nie ma sensu zmieniać/dodawać env lub konfiga, gdyż na ten moment tylko ten plik modyfikowany jest przez tłumaczenia i dodakowa abstrakcja tak prostej relacji tylko by zagmatwała sprawę.

sellmark replied:
robienie po funkcji pozwoliłoby przenieść także skrypt do tłumaczeń poza discord_bot, zwłasza jeśli chcemy go użyć w dota_bot? Bo nie wiem czy chcemy, ale chyba tak. 

potem zamiast self.getTranslations, miałbyś self.translationService.getTranslations("EN")

może za daleko myślę w projekcie z takim długiem technologicznym

PogSmok replied:
Jak będzie potrzeba to się doda, na razie nie widzę powodu. Do tego czasu może się struktura zmieni.

sellmark commented on app/balancer/management/commands/discord_bot.py:
````diff
@@ -461,21 +458,7 @@ async def whois_command(self, msg, **kwargs):
-            f'{player.description or ""}\n'
-            f'```'
-        )
+        await msg.channel.send(TRANSLATIONS[LANG]["whois_stats"].format(player.name, player.dota_mmr, dotabuff, player_url, player.ladder_mmr, player.score, player.rank_score, len(player.matches), wins, losses, "yes" if player.vouched else "no", Command.roles_str(player.roles), player.description or ""))
````
możemy to jakoś na linie rozbić? Jest bardzo nieczytelnie

sellmark commented on app/balancer/management/commands/discord_bot.py:
```diff
@@ -824,19 +781,13 @@ async def streak_command(self, msg, **kwargs):
 
         if not streaks:
-            await msg.channel.send(f'{self.player_mention(player)}: You didn\'t play a thing to have streak.')
+            await msg.channel.send(TRANSLATIONS[LANG]["no_streak"].format(player.name))
```
czy możemy pozostawić tutaj `self.player_mention(player)` ?

sellmark commented on app/balancer/management/commands/discord_bot.py:
```diff
@@ -1027,76 +943,61 @@ async def help_command(self, msg, **kwargs):
     async def set_name_command(self, msg, **kwargs):
         command = msg.content
         admin = kwargs['player']
+        #TODO HEREEEEEEEEEEE
```
chyba niepotrzebny komentarz

sellmark reviewed (changes requested):
Proszę zapoznaj się z komentarzami

PogSmok committed:
Create __init__.py

PogSmok committed:
Create PL.json

PogSmok committed:
Create EN.json

PogSmok committed:
Create translations.py

PogSmok committed:
Changes according to comments on pr

PogSmok committed:
Update EN.json

PogSmok committed:
Update PL.json

PogSmok committed:
Update EN.json

PogSmok committed:
Update PL.json

PogSmok committed:
Delete app/balancer/management/commands/EN.json

PogSmok committed:
Delete app/balancer/management/commands/PL.json

PogSmok committed:
Changed directory to app/balancer/

PogSmok committed:
Fixed queue_str error

PogSmok committed:
Translation update

PogSmok committed:
Translation Update

PogSmok committed:
Update discord_bot.py

Komendy usunięte:
- afk ping
- join
- leave
- join
- list

Komendy zmienione:
- registration help
- who is 
- vote kick
- recent 
- top/bot
- roles
- help
- adminhelp

TODO:
Skończyć format stringa "game_start"
